src/TesApi.Web/DoOnceAtStartUpService.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace TesApi.Web
+{
+    /// <summary>
+    /// Hosted service that executes one-time set up tasks at start up.
+    /// </summary>
+    public class DoOnceAtStartUpService : IHostedService
+    {
+        private readonly ILogger<DoOnceAtStartUpService> logger;
+        private readonly IServiceProvider serviceProvider;
+        /// <summary>
+        /// Hosted service that executes one-time set-up tasks at start up.
+        /// </summary>
+        /// <param name="serviceProvider"></param>
+        /// <param name="logger"></param>
+        public DoOnceAtStartUpService(IServiceProvider serviceProvider, ILogger<DoOnceAtStartUpService> logger)
+        {
+            ArgumentNullException.ThrowIfNull(serviceProvider);
+            ArgumentNullException.ThrowIfNull(logger);
+
+            this.serviceProvider = serviceProvider;
+            this.logger = logger;
+        }
+
+        /// <summary>
+        /// Executes start up tasks
+        /// </summary>
+        /// <param name="cancellationToken"></param>
+        public async Task StartAsync(CancellationToken cancellationToken)
+        {
+            using (logger.BeginScope("Executing Start Up tasks"))
+            {
+                try
+                {
+                    logger.LogInformation("Executing Configuration Utils Setup");
+                    await ExecuteConfigurationUtilsSetupAsync();
```

Comment:
What's the advantage of using a hosted service here, instead of just Task.Run() ?